apple/Sources/FerrostarMapLibreUI/Views/DynamicallyOrientingNavigationView.swift

```diff
@@ -106,8 +107,13 @@ public struct DynamicallyOrientingNavigationView: View, CustomizableNavigatingIn
                         showZoom: true,
                         onZoomIn: { camera.incrementZoom(by: 1) },
                         onZoomOut: { camera.incrementZoom(by: -1) },
-                        showCentering: !camera.isTrackingUserLocationWithCourse,
-                        onCenter: { camera = navigationCamera },
+                        cameraControlState: camera.isTrackingUserLocationWithCourse ? .showRouteOverview {
+                            if let overviewCamera = navigationState?.routeOverviewCamera {
+                                camera = overviewCamera
+                            }
+                        } : .showRecenter { // TODO: Third case when not navigating!
```

Comment:
TBD the best way to express this logic given the limits of view code; probably in an extension / helper function.